File: Algorithms/p376_Wiggle_Subsequence_2.cpp
class Solution {
public:
    int wiggleMaxLength(vector<int>& nums) {
        if(nums.empty()) return 0;
        int res = 1;
        int tail = 0; // 0:eq, 1:inc, 2:dec
        for(int i=1;i<nums.size();++i){
            if(nums[i]>nums[i-1]&&tail!=1){
                ++res;
                tail = 1;
            }else if(nums[i]<nums[i-1]&&tail!=2){
                ++res;
                tail = 2;
            }
        }
        return res;
    }
};

File: Algorithms/p376_Wiggle_Subsequence_2.py
class Solution(object):
    def wiggleMaxLength(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if not nums:
            return 0
        res = 1
        tail = 0 # 0:eq, 1:inc, 2:dec
        for i in range(1,len(nums)):
            if nums[i]>nums[i-1] and tail!=1:
                res += 1
                tail = 1
            elif nums[i]<nums[i-1] and tail!=2:
                res += 1
                tail = 2
        return res
